xarray/tests/test_distributed.py

```diff
@@ -25,28 +28,54 @@
 if has_h5netcdf:
     ENGINES.append('h5netcdf')
 
+NC_FORMATS = {'netcdf4': ['NETCDF3_CLASSIC', 'NETCDF3_64BIT_OFFSET',
+                          'NETCDF3_64BIT_DATA', 'NETCDF4_CLASSIC', 'NETCDF4'],
+              'scipy': ['NETCDF3_CLASSIC', 'NETCDF3_64BIT'],
+              'h5netcdf': ['NETCDF4']}
+TEST_FORMATS = ['NETCDF3_CLASSIC', 'NETCDF4_CLASSIC', 'NETCDF4']
+
+
+os.environ["HDF5_USE_FILE_LOCKING"] = "FALSE"
```

Comment:
thanks to @jakirkham for uncovering this doozy (https://github.com/dask/dask/pull/3179#issuecomment-366813164).

Reply:
If this turns up in our test suite, my guess is that this will also manifest itself to users. So it is probably also worth mentioning in the user facing documentation.

It might also be worth checking if this resolves the test failure with h5netcdf and autoclose (no worries if it doesn't).

Reply:
Glad to help. :)

Out of curiosity, where were you running into this issue (e.g. OS, file system, etc.)?

Reply:
@jakirkham - I'm also on OSX with a standard apple fusion file system.